src/index.ts

```diff
@@ -1,17 +1,29 @@
 import * as Vue from 'vue'
 import { VueClass } from './declarations'
-
-import { componentFactory } from './component'
+import { componentFactory, $internalHooks } from './component'
 
 export { createDecorator } from './util'
 
-export default function Component <U extends Vue>(options: Vue.ComponentOptions<U>): <V extends VueClass>(target: V) => V
-export default function Component <V extends VueClass>(target: V): V
-export default function Component <V extends VueClass>(options: Vue.ComponentOptions<any> | V): any {
+export interface ComponentDecorator {
+  <U extends Vue>(options: Vue.ComponentOptions<U>): <V extends VueClass>(target: V) => V
+  <V extends VueClass>(target: V): V
+
+  registerHooks (key: string[]): void
+}
+
+const Component: ComponentDecorator = function <V extends VueClass>(
+  options: Vue.ComponentOptions<any> | V
+): any {
   if (typeof options === 'function') {
     return componentFactory(options)
   }
   return function (Component: V) {
     return componentFactory(Component, options)
   }
+} as any
```

Comment:
I would suggest to use `namespace` trick to bypass type assertion. Here is an example https://github.com/HerringtonDarkholme/av-ts/blob/master/src/core.ts#L173